src/transformers/trainer_pt_utils.py

```diff
@@ -1395,3 +1395,211 @@ def get_lr(self):
+            " certain layers and recomputing them during a backward pass. Effectively, this trades extra"
+            " computation time for reduced memory usage."
+        },
+    )
+    state_dict_type: str = field(
+        default="FULL_STATE_DICT",
+        metadata={
+            "help": "State dict type to use. If a string, it must be one of `full_state_dict`, `local_state_dict`, or `sharded_state_dict`"
+        },
+    )
+    xla: bool = field(
+        default=False,
+        metadata={
+            "help": "Whether to use PyTorch/XLA Fully Sharded Data Parallel Training. This is an experimental feature"
+            " and its API may evolve in the future."
+        },
+    )
+    xla_fsdp_settings: Dict = field(
+        default=None,
+        metadata={
+            "help": "The value is a dictionary which stores the XLA FSDP wrapping parameters."
+            " For a complete list of options, please see [here]("
+            " https://github.com/pytorch/xla/blob/master/torch_xla/distributed/fsdp/xla_fully_sharded_data_parallel.py)."
+        },
+    )
+    xla_fsdp_grad_ckpt: bool = field(
+        default=False,
+        metadata={
+            "help": "Will use gradient checkpointing over each nested XLA FSDP wrapped layer. This setting can only be"
+            " used when the xla flag is set to true, and an auto wrapping policy is specified through"
+            " fsdp_min_num_params or fsdp_transformer_layer_cls_to_wrap."
+        },
+    )
+    xla_fsdp_v2: bool = field(
+        default=False,
+        metadata={"help": "Whether to use PyTorch/XLA Fully Sharded Data Parallel Training v2."},
+    )
+
+    @classmethod
+    def from_json_file(cls, json_file):
+        # Check if exists
+        open_file = io.open if os.path.exists(json_file) else open
+        with open_file(json_file, "r", encoding="utf-8") as f:
+            config_dict = json.load(f)
+        # for backward compatibility for keys starting with fsdp_ prefix
+        for k in list(config_dict.keys()):
+            if k.startswith("fsdp_"):
+                v = config_dict.pop(k)
+                config_dict[k[5:]] = v
+        # Check for keys and load sensible defaults
+        extra_keys = sorted(key for key in config_dict.keys() if key not in cls.__dataclass_fields__.keys())
+        if len(extra_keys) > 0:
+            raise ValueError(
+                f"The config file at {json_file} had unknown keys ({extra_keys}), please try upgrading your `transformers`"
+                " version or fix (and potentially remove these keys) from your config file."
+            )
+        return cls(**config_dict)
+
+    def to_dict(self):
+        return copy.deepcopy(self.__dict__)
+
+    def pop(self, key, default=None):
+        return self.__dict__.pop(key, default)
+
+    def __getitem__(self, key):
+        return self.__dict__.get(key)
+
+    def get(self, key, default=None):
+        return self.__dict__.get(key, default)
```

Comment:
### Code change comments


For backward compatibility with `trainer.py` and to limit changes to this file instead of propagating to other places where its being used as if its a Dict